File: august_long_challenge_2019/dstapls.cpp
/***
  *
 ***/
#include<iostream>

using namespace std;
int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int t;
	long long n, k, x;
	cin>>t;
	while(t--)
	{
		cin>>n>>k;
		x=(k%1000000007)*(k%1000000007);
		if(n%x==0)
			cout<<"NO\n";
		else
			cout<<"YES\n";
	}
	return 0;
}

File: august_long_challenge_2019/dstapls.py
'''***
    *
   ***'''
t = int(input())
while t:
	t-=1
	n,k = input().split()
	n = int(n)
	k = int(k)
	if (n % (k*k)) == 0:
		print("NO")
	else:
		print("YES")
